src/commands/unity-symbols/upload.ts

```diff
@@ -0,0 +1,255 @@
+import {MetricsLogger, getMetricsLogger} from '../../helpers/metrics'
+import {MultipartValue, UploadStatus} from '../../helpers/upload'
+import {DEFAULT_CONFIG_PATHS, performSubCommand, resolveConfigFromFileAndEnvironment} from '../../helpers/utils'
+import * as validation from '../../helpers/validation'
+import {checkAPIKeyOverride} from '../../helpers/validation'
+import {version} from '../../helpers/version'
+
+import * as dsyms from '../dsyms/upload'
+
+import {getUnityRequestBuilder, uploadMultipartHelper} from './helpers'
+import {IL2CPP_MAPPING_FILE_NAME, MappingMetadata, TYPE_IL2CPP_MAPPING, VALUE_NAME_IL2CPP_MAPPING} from './interfaces'
+import {
+  renderArgumentMissingError,
+  renderFailedUpload,
+  renderGitWarning,
+  renderMissingBuildId,
+  renderMissingIL2CPPMappingFile as renderMissingIl2CppMappingFile,
+  renderRetriedUpload,
+  renderUpload,
+} from './renderer'
+
+export class UploadCommand extends Command {
+  public static paths = [['unity-symbols', 'upload']]
+
+  public static usage = Command.Usage({
+    category: 'RUM',
+    description: 'Upload Unity symbol files to Datadog.',
+    details: `
+            This command will upload all iOS symbol files for Unity applications in order to symbolicate errors and
+            crash reports received by Datadog. This includes uploading dSYMs and IL2CPP mapping files.
+        `,
+    examples: [
+      [
+        'Upload all symbol files from the default location',
+        'datadog-ci unity-symbols upload --service-name com.datadog.example',
+      ],
+    ],
+  })
+
+  private disableGit = Option.Boolean('--disable-git', false)
+  private dryRun = Option.Boolean('--dry-run', false)
+  private configPath = Option.String('--config')
+  private maxConcurrency = Option.String('--max-concurrency', '20', {validator: validation.isInteger()})
+  private repositoryUrl = Option.String('--repository-url')
+  private symbolsLocation = Option.String('--symbols-location', './datadogSymbols')
+
+  private buildId?: string
+  private cliVersion = version
+  private config: Record<string, string> = {
+    datadogSite: 'datadoghq.com',
+  }
+  private gitData?: RepositoryData
+
+  public async execute() {
+    if (!(await this.verifyParameters())) {
+      return 1
+    }
+
+    this.config = await resolveConfigFromFileAndEnvironment(
+      this.config,
+      {
+        apiKey: process.env.DATADOG_API_KEY,
+        datadogSite: process.env.DATADOG_SITE,
+      },
+      {
+        configPath: this.configPath,
+        defaultConfigPaths: DEFAULT_CONFIG_PATHS,
+        configFromFileCallback: (configFromFile: any) => {
+          checkAPIKeyOverride(process.env.DATADOG_API_KEY, configFromFile.apiKey, this.context.stdout)
+        },
+      }
+    )
+
+    if (!this.disableGit) {
+      this.gitData = await this.getGitMetadata()
+    }
+
+    await this.performDsymUpload()
+    await this.performIl2CppMappingUpload()
+
+    return 0
```

Comment:
I have a few questions here.

1. From what I understand you don't use the return value from these 2 functions and instead always return `0`.
Is it on purpose so that a symbol files upload is not blocking?

2. The 2 uploads are not in parallel, is it on purpose as well?

Adding a suggestion if you want to make the calls in parallel and indicate failure if one of them failed to upload:

```suggestion
    const uploadResults = await Promise.all([
      this.performDsymUpload(),
      this.performIl2CppMappingUpload()
    ])

    return uploadResults.some((result) => result !== UploadStatus.Success) ? 1 : 0
```

Reply:
That's the correct behavior but actually I should write out a command summary.